tests/components/husqvarna_automower/test_init.py

```diff
@@ -128,20 +131,29 @@ async def test_websocket_not_available(
     freezer: FrozenDateTimeFactory,
 ) -> None:
     """Test trying reload the websocket."""
-    mock_automower_client.start_listening.side_effect = HusqvarnaWSServerHandshakeError(
-        "Boom"
+    mock_automower_client.auth.websocket_connect.side_effect = (
+        HusqvarnaWSServerHandshakeError("Boom")
     )
     await setup_integration(hass, mock_config_entry)
     assert "Failed to connect to websocket. Trying to reconnect: Boom" in caplog.text
     assert mock_automower_client.auth.websocket_connect.call_count == 1
-    assert mock_automower_client.start_listening.call_count == 1
     assert mock_config_entry.state is ConfigEntryState.LOADED
-    freezer.tick(timedelta(seconds=2))
+    reconnect_time = 2
+    for count in range(1, 945):
+        reconnect_time = min(reconnect_time * 2, MAX_WS_RECONNECT_TIME)
+        freezer.tick(timedelta(seconds=reconnect_time))
```

Comment:
We can't test it like this since we don't want to sleep for any time in tests. We could make the default reconnect time a constant in the integration to make it easier to patch, then patch that to zero with the `new=0` parameter, and then test that we don't hit the recursion limit.

Reply:
If we patch the time to 0, from the beginning of this test, then it depends on the machines performance, how many retries we've already have, until the loop starts. So the test is very flakey. Do we actually have to test it?

Reply:
I can just reset the mock..